src/c_api/value.cpp

```diff
@@ -610,6 +610,38 @@ kuzu_state kuzu_value_get_timestamp_tz(kuzu_value* value, kuzu_timestamp_tz_t* o
     return KuzuSuccess;
 }
 
+kuzu_state kuzu_value_get_decimal(kuzu_value* value, std::string* out_result) {
+    auto logical_type_id = static_cast<Value*>(value->_value)->getDataType().getLogicalTypeID();
+    auto physical_type_id = static_cast<Value*>(value->_value)->getDataType().getPhysicalType();
+    if (logical_type_id != LogicalTypeID::DECIMAL) {
+        return KuzuError;
+    }
+    switch (physical_type_id) {
+    case PhysicalTypeID::INT16:
+        *out_result = std::to_string(static_cast<Value*>(value->_value)->getValue<int16_t>());
+        return KuzuSuccess;
+
+    case PhysicalTypeID::INT32:
+        *out_result = std::to_string(static_cast<Value*>(value->_value)->getValue<int32_t>());
+        return KuzuSuccess;
+
+    case PhysicalTypeID::INT64:
+        *out_result = std::to_string(static_cast<Value*>(value->_value)->getValue<int64_t>());
+        return KuzuSuccess;
+
+    case PhysicalTypeID::INT128: {
+        auto int_value_representation = static_cast<Value*>(value->_value)->getValue<int128_t>();
+        *out_result = std::to_string(int_value_representation.low) +
+                      std::to_string(int_value_representation.high);
+        return KuzuSuccess;
+    }
+
+    default:
+        return KuzuError;
+    }
```

Comment:
You should NOT reinvent the logic of converting decimal to string. There is already a function `toString` you can use.

See: https://github.com/kuzudb/kuzu/blob/f15d8783345ee7b18cd23c09e07334b2540912c8/tools/java_api/src/jni/kuzu_java.cpp#L875-L879

and 
https://github.com/kuzudb/kuzu/blob/f15d8783345ee7b18cd23c09e07334b2540912c8/tools/python_api/src_cpp/py_query_result.cpp#L210-L213